### DISCUSSION TITLE
Create a texture exactly under the mouse?

### DISCUSSION THREAD

**Discussion by tseli0s** [parasyte/pixels#294] [Q&A] [answered]
# Create a texture exactly under the mouse?
I know this may not be the ideal place to ask, but I couldn't find much on the examples. If I want to create a new texture (4x4, each pixel a bit different) that is placed under the mouse cursor, how would I go about it? Currently I have a simple buffer sent every frame that clears the window and nothing more. 

Also I apologize if this question could be pretty obvious. I've only worked with SDL's renderer and raw OpenGL coordinates :P.

**Comment by parasyte** [comment, marked as answer]
I'm not sure exactly what your goal is. A "texture" has a very specific meaning in graphics APIs and positioning them requires applying a transformation matrix to it in a shader. `pixels` does not give you access to textures and shaders in the traditional graphics model. You are only given an array of bytes and it is up to you to plot pixels within the array. The reason for this is because `pixels` is designed for writing emulators and software renderers. I can't be certain if you intend to plot pixels, but it's safest to not make assumptions.

So, with that out of the way, do you actually want to plot pixels? Or do you want a higher-level abstraction that gives you control over concepts like textures and transformation matrices?

If you want to plot pixels, then you can use the [`Pixels::window_pos_to_pixel`](https://docs.rs/pixels/latest/pixels/struct.Pixels.html#method.window_pos_to_pixel) method to convert a mouse cursor position (e.g. from `winit`) to a pixel location that can be plotted within the pixel buffer. The `conway` example shows exactly how to do this: https://github.com/parasyte/pixels/blob/41063dbf4b105c45c7bb560f63b41459957faa85/examples/conway/src/main.rs#L76-L98

**Reply by tseli0s** [reply]
Pretty much the answer I was looking for. Also sorry for the confusion, I meant to say `SurfaceTexture`. Actually the only part I need is the `window_pos_to_pixel()` one, I didn't know what it did.
